### DISCUSSION TITLE
Is there a possibility, to add image upload/select on the right toolbar?

### DISCUSSION THREAD

**Discussion by wojto** [GrapesJS/grapesjs#5589] [Q&A]
I would like to prepare a component for image shown in modal, where I can choose the thumbnail image, and the big image, which will be open in modal.
For now, I only found the possibility to upload/choose image from the “image” component placed on the page, but I also need to have the possibility to select somewhere which image will be opened in modal (the best option will be to select this big image from the right toolbar).

Or maybe there is already a similar component/solution for modals?

**Comment by artf** [comment]
Here you can see an example of how you would add a [custom toolbar button](https://github.com/GrapesJS/tui-image-editor/blob/a9ad7aa8cd2015753285d6e7a058bfc3391eba1b/src/index.ts#L173-L190) for all image components
